overlord/servicestate/quota_control_test.go

```diff
@@ -105,20 +106,33 @@ func checkQuotaState(c *C, st *state.State, exp map[string]quotaGroupState) {
 		if len(expGrp.Snaps) != 0 {
 			c.Assert(grp.Snaps, DeepEquals, expGrp.Snaps)
 
-			// also check on the service file states
-			for _, sn := range expGrp.Snaps {
-				// meh assume all services are named svc1
-				slicePath := name
-				if grp.ParentGroup != "" {
-					slicePath = grp.ParentGroup + "/" + name
+			// also check on the service file states, but take into account
+			// that the services might be in separate sub-groups here. If it has
+			// sub-groups, assume those contain the services
+			if len(expGrp.SubGroups) == 0 {
+				for _, sn := range expGrp.Snaps {
+					// meh assume all services are named svc1
```

Comment:
It's a great idea, and actually how it already works. If a snap does not have a service named svc1, then the test will fail, because the expected systemctl output (which assumes svc1 naming) generated will not match the actual one generated by snapd.

Reply:
It's a great idea and actually how it works. If a snap does not have a service named svc1, then the check in `checkSvcAndSliceState` will fail and the test will fail. The test assumes the svc1 service unit exists, but it does assert on it. 
`c.Assert(svcFileName, testutil.FilePresent)`.